### PR TITLE
refactor: switch network modal to toast

### PR DESCRIPTION
**Pull request by fbwoolf**
## Description

This PR will hopefully fix several reported issues by removing the modal popup to switch networks. It is being replaced by a toast that simply lets the user know what network they are on. If the user is coming from a link which is on the opposite network, it will switch it automatically for them and show the toast.

For details refer to issue #481, #514, #518 (also check #360, #334, #349, #514)

image 

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Testing information

- Visit each page and switch networks to verify the toast is working as expected
- Open a transaction link from the wallet and see if switches network automatically

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -23,12 +28,27 @@ function ExplorerApp({
   apiServer,
   pageProps: { isHome, fullWidth, ...props },
 }: ExplorerAppProps) {
+  const router = useRouter();
+  const setUpdateNetworkIndex = useUpdateAtom(networkIndexState);
```
Comment (fbwoolf) [inline src/pages/_app.tsx:32]: Did not mean to use the word `Update` here in the name ...will remove.

File: src/common/hooks/use-filter-state.tsx [hunk 2]
```diff
@@ -9,7 +9,6 @@ import { usePrevious } from 'react-use';
 export const useFilterState = (key: 'sandbox' | 'txList') => {
   const [filter, setFilterState] = useAtom(filterState(key));
   const filterTypes = filter.types.filter(Boolean);
-  const previousFilterTypes = usePrevious(filterTypes);
```
[the remark this answers is not in the document]
Comment (fbwoolf) [reply src/common/hooks/use-filter-state.tsx:12]: @aulneau I put a debounce around the `handleClick` event of the `CheckableElement` but I'm still getting this error back, am I putting the debounce in the wrong place? 

Screen Shot 2021-09-03 at 10 49 19 AM

[the remark this answers is not in the document]
Comment (aulneau) [reply src/common/hooks/use-filter-state.tsx:12]: @fbwoolf can you share some code?

[the remark this answers is not in the document]
Comment (fbwoolf) [reply src/common/hooks/use-filter-state.tsx:12]: Yep, put this in `CheckableElement`:

```js
const handleClick = () => {
   onClick?.([type, !toggled]);
};

const debouncedHandleClick = React.useCallback(
   debounce(() => {
     handleClick();
   }, 1000),
   [handleClick]
);
```

[the remark this answers is not in the document]
Comment (fbwoolf) [reply src/common/hooks/use-filter-state.tsx:12]: I ended up being able to fix this by adding a `SafeSuspense` around the `TabbedTransactionList`.

**Comment by fbwoolf** [comment]
> While this does technically fix the issue, it is more of a bandaid than a true fix imo. I'd still recommend trying to check out how best to debounce the call before, you want to be able to use the last fetched query to display while the list is being suspended. This will remove the list completely and show the loading panel in place, which is less good from a UX perspective imo.

What makes this a bandaid in comparison to the use of Suspense and a loading fallback in other places in the explorer? I have tried debouncing the checkable elements in the filter list but I still get the promise error back from your toolkit, any suggestions on how to make the debounce work? I still don't totally understand why removing the `previousTypes` from the fallback makes it work either, why is that causing an uncaught promise error when just using the default types works?

**Comment by aulneau** [comment]
> What makes this a bandaid in comparison to the use of Suspense and a loading fallback in other places in the explorer?

The bandage is not the use of Suspense, but the fallback that has been passed. Currently as you have it implemented, this is what will happen:

- default list is shown with all types of transactions
- select filter to not show token transfers
- **list will disappear and show the generic loading screen**
- list is updated (sans token transfers)
- filter to additionally hide contract-calls
- **list will disappear and show the generic loading screen**
- list is updated (sans token + contract calls)

Where as the ideal state would be this (no loss of transaction list)

- default list is shown with all types of transactions
- select filter to not show token transfers
- **list is same while loading (by using a fallback component of the previous query data--all transactions)**
- list is updated (sans token transfers)
- select filter to additionally hide contract-calls (token transfers, contract-calls are filtered away)
- **list is loading and shows the list sans token transfers (the previous query data -- all tx aside from token transfers)**
- list is updated (sans token + contract calls)

Hope this helps -- i'm happy to take a look too, just wasn't able to look at it with all the RBF stuff happening.

**Comment by fbwoolf** [comment]
> 
> The bandage is not the use of Suspense, but the fallback that has been passed. Currently as you have it implemented, this is what will happen:
> 
Yep, that is correct. However, I rarely see the loading fallback unless I try real hard and click fast! Haha.
> 
> Hope this helps -- i'm happy to take a look too, just wasn't able to look at it with all the RBF stuff happening.

Sure thing, would love to see it working with the debounce you suggest. I must not be putting it in the right place.

**Comment by fbwoolf** [comment]
Added `Loading...` state to filter panel

Screen Shot 2021-09-07 at 7 05 14 PM

**Comment by fbwoolf** [comment]
@aulneau can you approve this and I'll move it into QA?

**Review by aulneau — APPROVED** [review]

**Comment by fbwoolf** [comment]
@timstackblock @Eshwari007 this is ready for QA. You should no longer see the modal asking you to switch networks. You should just see a toast popup that lets you know what network you are on when coming from a link or it will popup when you initiate switching networks in the app. Also, the filter issue should be resolved with this PR.

**Comment by Eshwari007** [comment]
@fbwoolf Its odd but yesterday I was able to view the  toast popup that lets you know what network you are when I select the networks dropdown.Today unfortunately am not able to view it.Does it only works when it comes from a link from activity tab?
FYI:explorer.stacks.com still has the switch modal.Only in https://explorer.staging.blockstack.xyz/ am not able to view it .

**Comment by Eshwari007** [comment]
Testing passed.

**Comment by fbwoolf** [comment]
> Testing passed.

Great, thank you!

**Pull request merged**